File: WOT-iOS/WOTData/WOTData/core/requests/Vehicles/WOTWEBRequestTankVehicles.m
//
//  WOTWEBRequestTankVehicles.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 6/19/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTWEBRequestTankVehicles.h"
#import "WOTDataDefines.h"

@implementation WOTWEBRequestTankVehicles

- (NSDictionary *)query {
    
    NSMutableDictionary *result = [[NSMutableDictionary alloc] init];
    result[WOTApiKeys.application_id] = [NSString valueOrSpaceString:self.hostConfiguration.applicationID];
    result[WOTApiKeys.fields] = [NSString valueOrSpaceString:[self.args escapedValueForKey:WOTApiKeys.fields]];
//    result[WOTApiKeys.tank_id] = [NSString valueOrSpaceString:[self.args escapedValueForKey:WOTApiKeys.tank_id]];
    return result;
}

- (NSString *)path {
    
    return @"/wot/encyclopedia/vehicles/";
}


- (NSString *)method {
    
    return @"POST";
}

@end

File: WOT-iOS/WOTData/Sources/webRequests/WOTWEBRequestTankVehicles.swift
//
//  WOTWEBRequestTankVehicles.swift
//  WOTData
//
//  Created by Pavel Yeshchyk on 1/14/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import WOTKit

@objc
public class WOTWEBRequestTankVehicles: WOTWEBRequest {
    override public var method: String { return "POST" }

    override public var path: String {
        return "/wot/encyclopedia/vehicles/"
    }
}

extension WOTWEBRequestTankVehicles: WOTModelServiceProtocol {
    @objc
    public class func modelClass() -> PrimaryKeypathProtocol.Type? {
        return Vehicles.self
    }

    @objc
    public func instanceModelClass() -> AnyClass? {
        return type(of: self).modelClass()
    }
}
